File: solutions/fsharp/darts/Darts.fs
module Darts

let score (x: double) (y: double): int =
    let distance = x * x + y * y |> sqrt
    match distance with
    | _ when distance <= 1 -> 10
    | _ when distance <= 5 -> 5
    | _ when distance <= 10 -> 1
    | _ -> 0

File: solutions/scala/darts/src/main/scala/Darts.scala
object Darts {
  def score(x: Double, y: Double): Int =
    Math.sqrt(x * x + y * y) match
      case distance if distance > 10 => 0
      case distance if distance > 5 => 1
      case distance if distance > 1 => 5
      case _ => 10
}